### ISSUE TITLE
Exception (java.lang.NoClassDefFoundError)  - RowService

### ISSUE THREAD

**Issue by daniyel** [Stratio/cassandra-lucene-index#88]
Hi.

I have cassandra 2.2.4 running on my docker container and lucene index plugin. Everything was working perfectly well until today, our container did not want to start. I have check what is the issue and i get the next error when i am starting the cassandra instance. And yes I have checked the lib folder from cassandra and the jar file is there. The jar package i get from http://central.maven.org/maven2/com/stratio/cassandra/cassandra-lucene-index-plugin/2.2.4.0/cassandra-lucene-index-plugin-2.2.4.0.jar. Cassandra is installed via 

```
apt-get install -y cassandra=2.2.4
```

and started as
```
cassandra -f
```

Loaded extensions:
```
...
INFO  09:20:14 Classpath: /etc/cassandra:/usr/share/cassandra/lib/ST4-4.0.8.jar:/usr/share/cassandra/lib/airline-0.6.jar:/usr/share/cassandra/lib/antlr-runtime-3.5.2.jar:/usr/share/cassandra/lib/cassandra-driver-core-2.2.0-rc2-SNAPSHOT-20150617-shaded.jar:/usr/share/cassandra/lib/cassandra-lucene-index-plugin-2.2.4.0.jar:/usr/share/cassandra/lib/commons-cli-1.1.jar:/usr/share/cassandra/lib/commons-codec-1.2.jar:/usr/share/cassandra/lib/commons-lang3-3.1.jar:/usr/share/cassandra/lib/commons-math3-3.2.jar:/usr/share/cassandra/lib/compress-lzf-0.8.4.jar:/usr/share/cassandra/lib/concurrentlinkedhashmap-lru-1.4.jar:/usr/share/cassandra/lib/crc32ex-0.1.1.jar:/usr/share/cassandra/lib/disruptor-3.0.1.jar:/usr/share/cassandra/lib/ecj-4.4.2.jar:/usr/share/cassandra/lib/guava-16.0.jar:/usr/share/cassandra/lib/high-scale-lib-1.0.6.jar:/usr/share/cassandra/lib/jackson-core-asl-1.9.2.jar:/usr/share/cassandra/lib/jackson-mapper-asl-1.9.2.jar:/usr/share/cassandra/lib/jamm-0.3.0.jar:/usr/share/cassandra/lib/javax.inject.jar:/usr/share/cassandra/lib/jbcrypt-0.3m.jar:/usr/share/cassandra/lib/jcl-over-slf4j-1.7.7.jar:/usr/share/cassandra/lib/jna-4.0.0.jar:/usr/share/cassandra/lib/joda-time-2.4.jar:/usr/share/cassandra/lib/json-simple-1.1.jar:/usr/share/cassandra/lib/libthrift-0.9.2.jar:/usr/share/cassandra/lib/log4j-over-slf4j-1.7.7.jar:/usr/share/cassandra/lib/logback-classic-1.1.3.jar:/usr/share/cassandra/lib/logback-core-1.1.3.jar:/usr/share/cassandra/lib/lz4-1.3.0.jar:/usr/share/cassandra/lib/metrics-core-3.1.0.jar:/usr/share/cassandra/lib/metrics-logback-3.1.0.jar:/usr/share/cassandra/lib/netty-all-4.0.23.Final.jar:/usr/share/cassandra/lib/ohc-core-0.3.4.jar:/usr/share/cassandra/lib/ohc-core-j8-0.3.4.jar:/usr/share/cassandra/lib/reporter-config-base-3.0.0.jar:/usr/share/cassandra/lib/reporter-config3-3.0.0.jar:/usr/share/cassandra/lib/sigar-1.6.4.jar:/usr/share/cassandra/lib/slf4j-api-1.7.7.jar:/usr/share/cassandra/lib/snakeyaml-1.11.jar:/usr/share/cassandra/lib/snappy-java-1.1.1.7.jar:/usr/share/cassandra/lib/stream-2.5.2.jar:/usr/share/cassandra/lib/super-csv-2.1.0.jar:/usr/share/cassandra/lib/thrift-server-0.3.7.jar:/usr/share/cassandra/apache-cassandra-2.2.4.jar:/usr/share/cassandra/apache-cassandra-thrift-2.2.4.jar:/usr/share/cassandra/apache-cassandra.jar:/usr/share/cassandra/stress.jar::/usr/share/cassandra/lib/jamm-0.3.0.jar
...
```

Error on starting the cassandra (irrelevant stuff from output are omitted):
```
...
INFO  09:20:18 Initializing Lucene index
Exception (java.lang.NoClassDefFoundError) encountered during startup: com/stratio/cassandra/lucene/service/RowService
java.lang.NoClassDefFoundError: com/stratio/cassandra/lucene/service/RowService
	at com.stratio.cassandra.lucene.Index.init(Index.java:103)
	at org.apache.cassandra.db.index.SecondaryIndexManager.addIndexedColumn(SecondaryIndexManager.java:290)
	at org.apache.cassandra.db.ColumnFamilyStore.<init>(ColumnFamilyStore.java:407)
	at org.apache.cassandra.db.ColumnFamilyStore.<init>(ColumnFamilyStore.java:354)
	at org.apache.cassandra.db.ColumnFamilyStore.createColumnFamilyStore(ColumnFamilyStore.java:535)
	at org.apache.cassandra.db.ColumnFamilyStore.createColumnFamilyStore(ColumnFamilyStore.java:511)
	at org.apache.cassandra.db.Keyspace.initCf(Keyspace.java:342)
	at org.apache.cassandra.db.Keyspace.<init>(Keyspace.java:270)
	at org.apache.cassandra.db.Keyspace.open(Keyspace.java:116)
	at org.apache.cassandra.db.Keyspace.open(Keyspace.java:93)
	at org.apache.cassandra.service.CassandraDaemon.setup(CassandraDaemon.java:256)
	at org.apache.cassandra.service.CassandraDaemon.activate(CassandraDaemon.java:529)
	at org.apache.cassandra.service.CassandraDaemon.main(CassandraDaemon.java:638)
Caused by: java.lang.ClassNotFoundException: com.stratio.cassandra.lucene.service.RowService
	at java.net.URLClassLoader$1.run(URLClassLoader.java:370)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 13 more
Caused by: java.util.zip.ZipException: invalid LOC header (bad signature)
	at java.util.zip.ZipFile.read(Native Method)
	at java.util.zip.ZipFile.access$1400(ZipFile.java:60)
	at java.util.zip.ZipFile$ZipFileInputStream.read(ZipFile.java:716)
	at java.util.zip.ZipFile$ZipFileInflaterInputStream.fill(ZipFile.java:419)
	at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:158)
	at sun.misc.Resource.getBytes(Resource.java:124)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:462)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	... 19 more
ERROR 09:20:18 Exception encountered during startup
java.lang.NoClassDefFoundError: com/stratio/cassandra/lucene/service/RowService
	at com.stratio.cassandra.lucene.Index.init(Index.java:103) ~[cassandra-lucene-index-plugin-2.2.4.0.jar:na]
	at org.apache.cassandra.db.index.SecondaryIndexManager.addIndexedColumn(SecondaryIndexManager.java:290) ~[apache-cassandra-2.2.4.jar:2.2.4]
	at org.apache.cassandra.db.ColumnFamilyStore.<init>(ColumnFamilyStore.java:407) ~[apache-cassandra-2.2.4.jar:2.2.4]
	at org.apache.cassandra.db.ColumnFamilyStore.<init>(ColumnFamilyStore.java:354) ~[apache-cassandra-2.2.4.jar:2.2.4]
	at org.apache.cassandra.db.ColumnFamilyStore.createColumnFamilyStore(ColumnFamilyStore.java:535) ~[apache-cassandra-2.2.4.jar:2.2.4]
	at org.apache.cassandra.db.ColumnFamilyStore.createColumnFamilyStore(ColumnFamilyStore.java:511) ~[apache-cassandra-2.2.4.jar:2.2.4]
	at org.apache.cassandra.db.Keyspace.initCf(Keyspace.java:342) ~[apache-cassandra-2.2.4.jar:2.2.4]
	at org.apache.cassandra.db.Keyspace.<init>(Keyspace.java:270) ~[apache-cassandra-2.2.4.jar:2.2.4]
	at org.apache.cassandra.db.Keyspace.open(Keyspace.java:116) ~[apache-cassandra-2.2.4.jar:2.2.4]
	at org.apache.cassandra.db.Keyspace.open(Keyspace.java:93) ~[apache-cassandra-2.2.4.jar:2.2.4]
	at org.apache.cassandra.service.CassandraDaemon.setup(CassandraDaemon.java:256) [apache-cassandra-2.2.4.jar:2.2.4]
	at org.apache.cassandra.service.CassandraDaemon.activate(CassandraDaemon.java:529) [apache-cassandra-2.2.4.jar:2.2.4]
	at org.apache.cassandra.service.CassandraDaemon.main(CassandraDaemon.java:638) [apache-cassandra-2.2.4.jar:2.2.4]
Caused by: java.lang.ClassNotFoundException: com.stratio.cassandra.lucene.service.RowService
	at java.net.URLClassLoader$1.run(URLClassLoader.java:370) ~[na:1.8.0_66]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362) ~[na:1.8.0_66]
	at java.security.AccessController.doPrivileged(Native Method) ~[na:1.8.0_66]
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361) ~[na:1.8.0_66]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[na:1.8.0_66]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331) ~[na:1.8.0_66]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[na:1.8.0_66]
	... 13 common frames omitted
Caused by: java.util.zip.ZipException: invalid LOC header (bad signature)
	at java.util.zip.ZipFile.read(Native Method) ~[na:1.8.0_66]
	at java.util.zip.ZipFile.access$1400(ZipFile.java:60) ~[na:1.8.0_66]
	at java.util.zip.ZipFile$ZipFileInputStream.read(ZipFile.java:716) ~[na:1.8.0_66]
	at java.util.zip.ZipFile$ZipFileInflaterInputStream.fill(ZipFile.java:419) ~[na:1.8.0_66]
	at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:158) ~[na:1.8.0_66]
	at sun.misc.Resource.getBytes(Resource.java:124) ~[na:1.8.0_66]
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:462) ~[na:1.8.0_66]
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73) ~[na:1.8.0_66]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368) ~[na:1.8.0_66]
	... 19 common frames omitted
```

As already stated everything was working fine, I could create index, fetch (query) the data until today when suddenly it "stopped" to work.

Best regards.

**Comment by ealonsodb** [comment]
Hi @daniyel. 

I dont know why you are getting that error. 

I have downloaded the jar pointed by you and indeed contains the class com.stratio.cassandra.lucene.service.RowService

I have also created a 3-node cluster with that jar using ccm and it is working properly. 

Apart from that, there is a new release version [2.2.4.1](https://github.com/Stratio/cassandra-lucene-index/tree/2.2.4.1) for cassandra 2.2.4 

Also, we have [docker containers](https://hub.docker.com/r/stratio/cassandra-lucene-index/), feel free to use them. 

Regards
